biz.aQute.bnd.exporters/src/aQute/bnd/exporter/subsystem/SubsystemExporter.java

```diff
@@ -50,99 +63,216 @@
 	@Override
 	public Map.Entry<String, Resource> export(String type, final Project project, Map<String, String> options)
 		throws Exception {
+
 		Jar jar = new Jar(".");
 
 		project.addClose(jar);
 
-		Manifest manifest = new Manifest();
-		manifest.getMainAttributes()
-			.putValue("Manifest-Version", "1.0");
-		manifest.getMainAttributes()
-			.putValue("Subsystem-ManifestVersion", "1");
+		EasArchivType esaArchivType = null;
+		Boolean storeBundles = null;
+		String archiveContent = project.get("-archiveContentType");
 
-		List<Container> distro = project.getBundles(Strategy.LOWEST, project.getProperty(Constants.DISTRO),
-			Constants.DISTRO);
-		List<File> distroFiles = getBundles(distro, project);
-		List<File> files = getBundles(project.getRunbundles(), project);
+		esaArchivType = EasArchivType.byParameter(archiveContent);
+		if (esaArchivType == null) {
+			esaArchivType = EasArchivType.CONTENT;
+		}
 
-		MultiMap<String, Attrs> imports = new MultiMap<>();
-		MultiMap<String, Attrs> exports = new MultiMap<>();
-		Parameters requirements = new Parameters();
-		Parameters capabilities = new Parameters();
+		Collection<Container> requireBundles = getRequiredBundles(project);
+		Collection<Container> runBundles = project.getRunbundles();
+
+		Collection<Container> bundles = null;
+		switch (esaArchivType) {
+			case NONE :
+				bundles = requireBundles;
+				storeBundles = false;
+				break;
+
+			case CONTENT :
+
+				bundles = requireBundles;
+				storeBundles = true;
+				break;
+
+			case ALL :
+
+				bundles = runBundles;
+				storeBundles = true;
+				break;
+
+			default :
+
+				throw new IllegalArgumentException("Unknown EsaArchivType: " + esaArchivType);
```

Comment:
I don't think we want to throw exceptions from a plugin. Better to put an error on the project and return.